tutorials/persistance-comparison-series/BinaryWriterReader/binary_reader_writer.md

```diff
@@ -0,0 +1,426 @@
+
+<img src="images/07_hit_count_extended_editor.jpg" alt="File BinaryReaderWriterExampleExtended" width=""/>
+
+After stopping the game and therefore saving the data, a new file `BinaryReaderWriterExampleExtended` should exist in your project folder. Have a look at it. It should look something like this:
+
+<img src="images/08_hit_count_extended_file.png" alt="File BinaryReaderWriterExampleExtended" width=""/>
+
+The three hit counters can be seen in there and correspond to the values in the inspector:
+
+- `0f` == 15
+- `0c` == 12
+- `05` == 5
+
+Last but not least, let's look at how to load the file again when starting the game (14):
+
+```cs
+private void Start() // 14
+{
+    // Check if the file exists to avoid errors when opening a non-existing file.
+    if (File.Exists(HitCountFile)) // 15
+    {
+        // Open a stream to the file that the `BinaryReader` can use to read data.
+        // They need to be disposed at the end, so `using` is good practice
+        // because it does this automatically.
+        using FileStream fileStream = File.Open(HitCountFile, FileMode.Open); // 16
+        using BinaryReader binaryReader = new(fileStream); // 17
+        hitCountUnmodified = binaryReader.ReadInt32(); // 18
+        hitCountShift = binaryReader.ReadInt32(); // 18
+        hitCountControl = binaryReader.ReadInt32(); // 18
+    }
+}
+```
+
+First, we check if the file even exists (15). If we ever saved data before, this should be the case. If it exists, we read the databy creating a `FileStream` again (16) and opening a `BinaryReader` with it (17). Similar to writing with `Write()` (on the `BinaryWriter`), we use `ReadInt32()` (18) to read an `integer`. We do this three times since we saved them all individually.
+
+Note that knowing the structure of the file is necessary here. If we saved an `integers`, a `boolean`, and a `string`, we would have to use `ReadInt32()`, `ReadBoolean()`, and `ReadString()`.
+
+The more complex data gets, the more complicated it will be to make sure there are no mistakes in the structure when reading or writing it. Different types, adding and removing variables, changing the structure. The more data we want to add to this file, the more it makes sense to think about alternatatives. For this tutorial, we will stick with the `BinaryReader` and `BinaryWriter` and see what we can do to decrease the complexity a bit when adding more data.
+
+One of those options will be shown in the next section.
+
+## More complex data
+
+(See `BinaryReaderWriterExampleJson.cs` in the repository for the finished version.)
+
+JSON is a very common approach when saving structured data. It's easy to use and there are frameworks for almost every language. The .NET framework provides a [`JsonSerializer`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=net-6.0). Unity has its own version of it: [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html).
```

Comment:
> It's easy to use and there are frameworks for almost every language.

I'm not sure frameworks is a good word here. I'd use the generic word "library" or if you want to be more precise "encoder/decoder" maybe?

Reply:
Fair point. I'm too used to saying SDK everywhere, but libs is deffo a better description here. :+1: 
ac15b8e9decad9f95666d5e37b811f651dc4a7dc